[seanpzk] iP

seanpzk opened the pull request:
### ChadBob
> "Thanks for reviewing my iP! Check out this [guy](https://github.com/), he is better :)"  

ChadBob frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here.](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is __FREE__!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
  
If you are a Java  programmer, you can use it to practice Java too. Here's the  `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

seanpzk committed:
Renamed Duke to ChadBob. Now main simply greets user and exits.

seanpzk committed:
Added echo functionality

seanpzk committed:
Added add and list functionality

seanpzk committed:
Added Task class.

seanpzk committed:
Added marking functionality for chatbot.

seanpzk committed:
Added javadoc comments for Duke and Task class

seanpzk committed:
Renamed Task's method

seanpzk committed:
Added Deadline, Event and ToDo classes

seanpzk committed:
Modified input.txt and EXPECTED.txt for Automated Text UI Testing.

seanpzk committed:
Added InvalidUserInputException exception.

seanpzk committed:
Added error handling functionality. Code reorganised for error handling.

seanpzk committed:
Added Task Manager class to handle collection of tasks.

seanpzk committed:
Added more commands to test in input.txt

seanpzk committed:
Restructured Duke.java for Task Manager. Also, fixed minor bug for testing.

seanpzk committed:
Added delete functionality.

seanpzk committed:
Add ResponseGen class.

There are a lot of different responses generated to inform user of the
change hardcoded inside Duke.java. ResponseGen will now handle the
different responses to be generated. It will return a String for duke to
print out to users.

seanpzk committed:
Update input.txt for testing

seanpzk committed:
Stop tracking src/main/data

seanpzk committed:
Update Task and its subclasses' constructors

seanpzk committed:
Add FileManager class to save task data in ChadBob

seanpzk committed:
Add Save functionality in Duke.

Tasks data are cleared previously when chatbot closes.

Save functionality automatically saves tasks data and chatbot loads the
tasks data upon start up.

seanpzk committed:
Merge branch 'branch-level-7'

Add save functionality in chatbot.
Add classes to better handle files and responses to users.

seanpzk committed:
Move printing job from Duke to ResponseGen

seanpzk committed:
Fix bug on additional File loading to TaskManager

seanpzk committed:
Teach chatbot how to understand dates and times.

seanpzk committed:
Merge branch 'branch-Level-8'

Teach chatbot how to understand dates and times.
DateTimeDetector class: Use to detect date and time and format
accordingly.
Bug fix: Fix updating Task Manager with saved file excessively.

seanpzk committed:
Rename classes name to align with iP requirements

seanpzk committed:
Add Parser class to deal with user input

seanpzk committed:
Merge branch 'A-MoreOOP'

Implement more OOP to the project.

seanpzk committed:
Update format of time stored in file

seanpzk committed:
Move files into packages

seanpzk committed:
Merge branch 'branch-A-Packages'

Move classes into packages

seanpzk committed:
Update runtest.bat after packaging classes

seanpzk committed:
Merge branch 'add-gradle-support'

Add gradle support

seanpzk committed:
Add Gradle to project

seanpzk committed:
Add JUnit test cases

seanpzk committed:
Add JavaDoc comments

seanpzk committed:
Tweak code to comply coding standard

seanpzk committed:
Add Find functionality

seanpzk committed:
Merge branch 'branch-Level-9'

Add Find functionality.

seanpzk committed:
Merge branch 'branch-A-JavaDoc'

Add JavaDoc comments.

seanpzk committed:
Merge branch 'branch-A-CodingStandard'

Tweak code to comply coding standard

seanpzk committed:
Touch up on coding standard and JavaDoc after merging 3 branches

seanpzk committed:
Fix importing all java.util

seanpzk committed:
Implement GUI

seanpzk committed:
Fix warning from version mismatch

seanpzk committed:
Merge branch 'branch-Level-10'

Implemented GUI for Duke

seanpzk committed:
Clear out unwanted commented code in build.gradle

kohkaijie commented on src/main/java/duke/userio/Parser.java:
```diff
@@ -0,0 +1,121 @@
+     * @throws InvalidUserInputException User input does not generate any specific response.
+     */
+    public void listen(String input) throws InvalidUserInputException {
+        if (input.equals("bye")) {
```
Perhaps you can use switch-case statements to reduce the use of if-else statements?

kohkaijie commented on src/main/java/duke/filemanagement/DateTimeDetector.java:
```diff
@@ -0,0 +1,92 @@
+    public String format(String date) {
+        LocalDate d;
+        if (date.contains(" ")) {
+            // date and time given
```
Great use of comments to give clarity on how the various date formats are checked!

kohkaijie commented on src/main/java/duke/userio/Ui.java:
```diff
@@ -0,0 +1,166 @@
+package duke.userio;
+
+import duke.task.*;
```
Perhaps you should avoid the use of asterisks to import the whole task package but import them individually when necessary instead?

kohkaijie commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,59 @@
+        this.isDone = false;
+    }
+
+    public boolean containString(String string) {
```
Perhaps you can name it containsString instead so that it is clearer it is a boolean?

kohkaijie commented on src/main/java/duke/userio/Parser.java:
```diff
@@ -0,0 +1,121 @@
+            updateSaveFile();
+        } else if (input.contains("find")) {
+            String inputToFind = input.substring(6);
+            TaskList tempTL = new TaskList();
```
Perhaps you can name it tempTl instead so that the abbrievation is not full uppercase?

kohkaijie reviewed (commented):
Overall great work! Great use of OOP to divide into multiple classes, variable naming all achieve the standard and line spacing and indentation is precise as well! Code is easy to understand. Keep up the good work!

seanpzk committed:
Change echoing functionality of bot to proper response from bot

seanpzk committed:
Implement switch case in Parser for better readability

seanpzk committed:
Fix bugs on GUI

Fix bug #1 where application did not stop after user input "bye".
Fix bug #2 where previous tasks are not loaded and user always start
from a fresh task list.

seanpzk committed:
Improve code quality

There are poor coding practices across the files such as poor naming
convention with boolean or deeply nested code, making the code hard to
read.

Corrected mistakes pointed out from PRs and own thorough analysis of my
code.

seanpzk committed:
Fix minor issue on coding practice

seanpzk committed:
Add assertions

seanpzk committed:
Merge pull request #2 from seanpzk/branch-A-CodeQuality

Improved Code Quality

seanpzk committed:
Merge branch 'branch-A-Assertions'

Add assertions to handle errors

seanpzk committed:
Add update functionalities to the different tasks

seanpzk committed:
Add update functionality to bot

seanpzk committed:
Merge branch 'branch-C-update'

TaskList class: Do not support editing of the task.

Only way to edit a task for a user is to delete the task and reinsert a
new task, making it inconvenient for them.

Added update functionality to update different attributes of a Task
object.

Parser will take input on update and pass the details to TaskList as
TaskList handles Tasks. The Tasks update function are called and updated
accordingly.

seanpzk committed:
Merge branch 'branch-C-update'

Added update functionality to update task details

seanpzk committed:
Merge branch 'master' of https://github.com/seanpzk/ip

seanpzk committed:
Add Github workflow

seanpzk committed:
Merge branch 'branch-A-CI'

Added Github workflow to run automated checks

seanpzk committed:
Add product screenshot

seanpzk committed:
Fix bug in DateTimeFormatter

seanpzk committed:
Fix minor bugs

seanpzk committed:
Update User Guide

seanpzk committed:
Merge branch 'branch-A-UserGuide'

Update User Guide

seanpzk committed:
Add JavaDoc comments

seanpzk committed:
Add test cases for DateTimeDetector

seanpzk committed:
Add more information for handling exception

seanpzk committed:
Fix gradle checkstyle errors

seanpzk committed:
Remove printing messages

seanpzk committed:
Fix minor bug on reading path

seanpzk committed:
Improve code quality

seanpzk committed:
Remove emojis that caused rendering error

seanpzk committed:
Fix cross platforms problems on Linux
